src/plugins/data/public/utils/table_inspector_view/components/data_table.tsx

```diff
@@ -35,40 +35,59 @@ import { i18n } from '@kbn/i18n';
 
 import { DataDownloadOptions } from './download_options';
 import { DataViewRow, DataViewColumn } from '../types';
-import { TabularData } from '../../../../common/adapters/data/types';
 import { IUiSettingsClient } from '../../../../../../core/public';
+import { Datatable } from '../../../../../expressions/common/expression_types/specs';
+import { FieldFormatsStart } from '../../../field_formats';
+import { UiActionsStart } from '../../../../../ui_actions/public';
 
 interface DataTableFormatState {
   columns: DataViewColumn[];
   rows: DataViewRow[];
 }
 
 interface DataTableFormatProps {
-  data: TabularData;
+  data: Datatable;
   exportTitle: string;
   uiSettings: IUiSettingsClient;
-  isFormatted?: boolean;
+  fieldFormats: FieldFormatsStart;
+  uiActions: UiActionsStart;
+}
+
+interface RenderCellArguments {
+  table: Datatable;
+  columnIndex: number;
+  rowIndex: number;
+  formattedValue: string;
+  uiActions: UiActionsStart;
 }
 
 export class DataTableFormat extends Component<DataTableFormatProps, DataTableFormatState> {
   static propTypes = {
     data: PropTypes.object.isRequired,
     exportTitle: PropTypes.string.isRequired,
     uiSettings: PropTypes.object.isRequired,
-    isFormatted: PropTypes.bool,
+    fieldFormats: PropTypes.object.isRequired,
+    uiActions: PropTypes.object.isRequired,
   };
 
   csvSeparator = this.props.uiSettings.get('csv:separator', ',');
   quoteValues = this.props.uiSettings.get('csv:quoteValues', true);
   state = {} as DataTableFormatState;
 
-  static renderCell(dataColumn: any, value: any, isFormatted: boolean = false) {
+  static renderCell({
+    table,
+    columnIndex,
+    rowIndex,
+    formattedValue,
+    uiActions,
+  }: RenderCellArguments) {
+    const column = table.columns[columnIndex];
     return (
       <EuiFlexGroup responsive={false} gutterSize="s" alignItems="center">
-        <EuiFlexItem grow={false}>{isFormatted ? value.formatted : value}</EuiFlexItem>
+        <EuiFlexItem grow={false}>{formattedValue}</EuiFlexItem>
```

Comment:
While we're in the neighborhood, can we take a second to improve the accessibility here?

Below, we set an `aria-label` for `EuiButtonIcon` to `"Filter for value"` which will lead to many, many buttons all with the same name but perform different actions. 

To improve this, can we change the `aria-label` to read `"Filter for value: {formattedValue}"`. This way, each unique button has a unique name.

Reply:
i'll keep this for a separate PR this one already got quite big.

Reply:
Follow up issue here: https://github.com/elastic/kibana/issues/85614